master/internal/db/postgres_experiments.go

```diff
@@ -474,48 +473,17 @@ SELECT t.id FROM (
 	return trials, err
 }
 
-func timeToFloat(t time.Time) float64 {
-	// If time.Time is the empty value, UnixNano will return the farthest back
-	// timestamp a float can represent, which is some large negative value.
-	if t.IsZero() {
-		return 0
-	}
-	return float64(t.UnixNano()) / 1e9
-}
-
-func scanMetricsSeries(rows *sql.Rows, metricSeriesBatch, metricSeriesTime []lttb.Point) (
-	[]lttb.Point, []lttb.Point, time.Time,
-) {
-	var maxEndTime time.Time
-	for rows.Next() {
-		var batches uint
-		var value float64
-		var endTime time.Time
-		err := rows.Scan(&batches, &value, &endTime)
-		if err != nil {
-			// Could be a bad metric name, sparse metric, nested type, etc.
-			continue
-		}
-		metricSeriesBatch = append(metricSeriesBatch, lttb.Point{X: float64(batches), Y: value})
-		metricSeriesTime = append(metricSeriesTime, lttb.Point{X: timeToFloat(endTime), Y: value})
-		if endTime.After(maxEndTime) {
-			maxEndTime = endTime
-		}
-	}
-	return metricSeriesBatch, metricSeriesTime, maxEndTime
-}
-
 // TrainingMetricsSeries returns a time-series of the specified training metric in the specified
 // trial.
 func (db *PgDB) TrainingMetricsSeries(trialID int32, startTime time.Time, metricName string,
-	startBatches int, endBatches int) (metricSeriesBatch, metricSeriesTime []lttb.Point,
-	maxEndTime time.Time, err error,
+	startBatches int, endBatches int) (rows *sql.Rows, err error,
```

Comment:
seems like returning `*sql.Rows` here makes it harder to understand what this function does.

Reply:
this will be updated in: https://github.com/determined-ai/determined/pull/5946#discussion_r1101693957